upstream-builder.Dockerfile

```diff
@@ -5,7 +5,7 @@ WORKDIR /build
 
 COPY . .
 RUN make static
-RUN GRPC_HEALTH_PROBE_VERSION=v0.4.28 && \
+RUN GRPC_HEALTH_PROBE_VERSION=v0.4.37 && \
```

Comment:
We should fix the FROM/AS case parity in this one too
i.e. line 1 goes to `FROM golang:1.23-alpine AS builder`